libcst/metadata/scope_provider.py

```diff
@@ -217,24 +217,12 @@ def _index(self) -> int:
         return self.__index
 
     def get_qualified_names_for(self, full_name: str) -> Set[QualifiedName]:
-        scope = self.scope
-        name_prefixes = []
-        while scope:
-            if isinstance(scope, ClassScope):
-                name_prefixes.append(scope.name)
-            elif isinstance(scope, FunctionScope):
-                name_prefixes.append(f"{scope.name}.<locals>")
-            elif isinstance(scope, ComprehensionScope):
-                name_prefixes.append("<comprehension>")
-            elif not isinstance(scope, (GlobalScope, BuiltinScope)):
-                raise Exception(f"Unexpected Scope: {scope}")
-
-            scope = scope.parent if scope.parent != scope else None
-
-        parts = [*reversed(name_prefixes)]
-        if full_name:
-            parts.append(full_name)
-        return {QualifiedName(".".join(parts), QualifiedNameSource.LOCAL)}
+        return {
+            QualifiedName(
+                ".".join(filter(None, [self.scope._name_prefix, full_name])),
```

Comment:
Neither of these can be `None`, and this is a pretty hot function. This should be much faster
```suggestion
                f"{self.scope._name_prefix}.{full_name}",
```

Reply:
The filter(None,...) also removes empty strings, which is what I need here. Happy to consider other ways to do it.